relayer/listener.go

```diff
@@ -259,6 +213,8 @@ func (lstnr *Listener) ProcessLogs(ctx context.Context) error {
 				return fmt.Errorf("failed to catch up on historical blocks")
 			}
 		case blockHeader := <-lstnr.Subscriber.Headers():
+			// Parse the logs in the block, and group by application relayer
```

Comment:
IIUC, we still make a blocking RPC call per block here to fetch all of the warp logs from it. As long as the latency of that request is significantly less that the block interval I don't think it will be a bottleneck, but would it be difficult to put it in its own goroutine either way? Something like a `go lstnr.processBlockHeader(blockHeader)` that encapsulates the logic here possibly.

Reply:
That shouldn't be difficult, but I don't know how much we'd gain. As you said, adding concurrency here would only make sense if the RPC rtt is comparable to the block interval, which is not the expected case (~100ms vs ~1s). It is an interesting optimization opportunity though. I will create a ticket enumerating possible optimization targets and include that there.